OpenRA.Game/Traits/World/DefaultOrderGenerator.cs

```diff
@@ -0,0 +1,31 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2021 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Orders;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class DefaultOrderGeneratorInfo : TraitInfo
+	{
+		public override object Create(ActorInitializer init) { return new DefaultOrderGenerator(); }
+	}
+
+	public class DefaultOrderGenerator : IDefaultOrderGenerator
```

Comment:
`DefaultOrderGeneratorFactory`

Reply:
This is trait. Trait will be named DefaultOrderGeneratorFactory? is this ok @pchote ?

Reply:
For me

World:
          Selection:
          DefaultOrderGeneratorFactory:

in world.yaml will look weird. but If it is ok, I'll rename it here

Reply:
All changes done